### PR TITLE
AZ-S2D

### PR DESCRIPTION
**Pull request by justinmlangford**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by ChrisEby** [comment]
What is the status of this pull?

**Comment by MSBrett** [comment]
This PR adds support for SQL 2016 FCI clusters per https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sql/virtual-machines-windows-portal-sql-create-failover-cluster

File: sql-server-2016-fci-existing-vnet-and-ad/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
++ 	The images used to create this deployment are
+	+ 	SQL Server 2016 SP1 and Windows Server 2016 Datacenter Edition - Latest Image
+
++	To successfully deploy this template, be sure that the subnet to which the storage nodes are being deployed already exists on the specified Azure virtual network, AND this subnet should be defined in Active Directory Sites and Services for the appropriate AD site in which the closest domain controllers are configured.
+
++ SPECIAL THANKS to <a href="https://github.com/mmarch">@mmarch</a> on code contributions for dynamic data disk selection nested templates!
++ SPECIAL THANKS to <a href="https://github.com/robotechredmond">@robotechredmond</a> for the Windows S2D template this is based on!
+
+## Deploying Sample Templates
```
Comment (bmoore-msft) [inline sql-server-2016-fci-existing-vnet-and-ad/README.md]: move this section to the top as in the sample

File: sql-server-2016-fci-existing-vnet-and-ad/dsc/ConfigSFCI.ps1 [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            RebootNodeIfNeeded = $true
+        }
+        
+        WindowsFeature FC
+        {
+            Name = "Failover-Clustering"
+            Ensure = "Present"
+        }
+
+		WindowsFeature FailoverClusterTools
```
Comment (bmoore-msft) [inline sql-server-2016-fci-existing-vnet-and-ad/dsc/ConfigSFCI.ps1]: can you clean up formatting in the file - indentation is off in a number of places

File: sql-server-2016-fci-existing-vnet-and-ad/nestedtemplates/deploy-cluster.json [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+    "ilbFEConfigID1": "[concat(variables('ilbID'),'/frontendIPConfigurations/',variables('ilbFE'))]",
+    "ilbBEAddressPoolID": "[concat(variables('ilbID'),'/backendAddressPools/',variables('ilbBE'))]",
+    "ilbProbe1": "[concat(parameters('namePrefix'),'ilbp1')]",
+    "ilbProbeID1": "[concat(variables('ilbID'),'/probes/',variables('ilbProbe1'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/availabilitySets",
+      "name": "[variables('vmAvailabilitySetName')]",
+      "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline sql-server-2016-fci-existing-vnet-and-ad/nestedtemplates/deploy-cluster.json]: update apiVersion to non-preview version

File: sql-server-2016-fci-existing-vnet-and-ad/nestedtemplates/diskSelection.json [hunk 4]
```diff
@@ -0,0 +1,263 @@
+{
```
Comment (bmoore-msft) [inline sql-server-2016-fci-existing-vnet-and-ad/nestedtemplates/diskSelection.json]: remove this deployment and use a property copy loop

File: sql-server-2016-fci-existing-vnet-and-ad/nestedtemplates/newVM.json [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                    },
+                    "diskNamingPrefix": {
+                        "value": "[concat(parameters('vmName'), '-data')]"
+                    }
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline sql-server-2016-fci-existing-vnet-and-ad/nestedtemplates/newVM.json]: update apiVersion

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**